### DISCUSSION TITLE
PtyPy improvements

### DISCUSSION THREAD

**Discussion by pierrethibault** [ptycho/ptypy#457] [Announcements]
The first draft of planned ptypy improvements can be found on this google doc:

https://docs.google.com/document/d/1jp1tqEqPJxw4isc4FTKXCqzsOrgcuiBkq4XjMHWfgMc/edit?usp=sharing

Anyone planning to participate is welcome to take a look and comment or edit. We will then break these general goals in smaller pieces to make a roadmap.

**Comment by jcesardasilva** [comment]
This is very nice, Pierre, and very important. I put some suggestions in the google doc you shared with us. I was wondering if it is possible to make clearer in the documentation how to properly use Ptypy for near-field ptychography, given that the tutorial presented is for Far-field ptychography. Is there any interested in including Fourier Ptychography in the package?

**Comment by pierrethibault** [comment]
Thanks Julio. I think @stefanoschalkidis will have at least templates to suggest for near-field ptychography.

Including Fourier ptychography would be very nice, thanks for mentioning. As for pyFFTW, I looked into this a long time ago and thought that scipy's FFTpack was just as fast?

**Comment by aaron-parsons** [comment]
Hi Pierre,

I also added some comments. From my view, I think adding unit tests is very important to make this a long lasting package. We should make a simple test framework so that contributors can build a test with very little effort/ time. This means that if we make framework changes (ie moving to gpu) we can test the whole structure straight away.

The tests also would work nicely as example code for users.

Aaron

## 

This e-mail and any attachments may contain confidential, copyright and or privileged material, and are for the use of the intended addressee only. If you are not the intended addressee or an authorised recipient of the addressee please notify us of receipt by returning the e-mail and do not use, copy, retain, distribute or disclose the information in or attached to the e-mail.
Any opinions expressed within this e-mail are those of the individual and not necessarily of Diamond Light Source Ltd. 
Diamond Light Source Ltd. cannot guarantee that this e-mail or any attachments are free from viruses and we cannot accept liability for any damage which you may sustain as a result of software viruses which may be transmitted in or with the message.
Diamond Light Source Limited (company no. 4375679). Registered in England and Wales with its registered office at Diamond House, Harwell Science and Innovation Campus, Didcot, Oxfordshire, OX11 0DE, United Kingdom
